### PR TITLE
Entropy filter fixes

### PR DESCRIPTION
**Pull request by FreddieWitherden**

### CODE REVIEW

File: pyfr/solvers/euler/kernels/entropy.mako [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
 
-<% inf = 1e20 %>
 <%pyfr:macro name='compute_entropy' params='u, d, p, e'>
     d = u[0];
     fpdtype_t rcpd = 1.0/d;
     fpdtype_t E = u[${nvars - 1}];
 
     // Compute the pressure
-    p = ${c['gamma'] - 1}*(E - 0.5*rcpd*(${pyfr.dot('u[{i}+1]', i=ndims)}));
+    p = ${c['gamma'] - 1}*(E - 0.5*rcpd*(${pyfr.dot('u[{i} + 1]', i=ndims)}));
```
Comment (tarikdzanic) [inline pyfr/solvers/euler/kernels/entropy.mako:9]: `i=(1, ndims+1)` ?

File: pyfr/backends/base/backend.py [hunk 2]
```diff
@@ -37,6 +37,8 @@ def __init__(self, cfg):
 
         # Convert to a NumPy data type
         self.fpdtype = np.dtype(prec).type
+        self.fpdtype_eps = np.finfo(self.fpdtype).eps
```
Comment (tarikdzanic) [inline pyfr/backends/base/backend.py:40]: Don't know if we still need eps since we solve for `exp(\zeta) \in [0,1]` now

Comment (FreddieWitherden) [reply pyfr/backends/base/backend.py:40]: Yeah, it's not currently used.  Although a couple of pending PR's want it, so decided to add whilst I was working on the code.

File: pyfr/solvers/euler/kernels/entropy.mako (hunk 1 above)
Comment (FreddieWitherden) [reply pyfr/solvers/euler/kernels/entropy.mako:9]: Will update.

**Pull request merged**